tests/test_java.py

```diff
@@ -536,3 +536,28 @@ def test_java_attach_socket_missing(
         profile = snapshot_one_profile(profiler)
         assert len(profile.stacks) == 1
         assert next(iter(profile.stacks.keys())) == "java;[Profiling error: exception JattachSocketMissingException]"
+
+
+# we know what messages to expect when in container, not on the host Java
+@pytest.mark.parametrize("in_container", [True])
+def test_java_jattach_async_profiler_log_output(
+    tmp_path: Path,
+    application_pid: int,
+    caplog: LogCaptureFixture,
+) -> None:
+    """
+    Tests that AP log is collected and logged in gProfiler's log.
+    """
+    caplog.set_level(logging.DEBUG)
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        duration=1,
+    ) as profiler:
+        snapshot_one_profile(profiler)
+
+        log_records = list(filter(lambda r: r.message == "async-profiler log", caplog.records))
+        assert len(log_records) == 2  # start,stop
+        # start
+        assert log_records[0].gprofiler_adapter_extra["ap_log"] == ""  # type: ignore
```

Comment:
Both of them print empty log, I wonder if that's enough for the tests or if I should find something else that triggers real logs written so that I could test it

Reply:
Improved